roles/devbind/tasks/main.yml

```diff
@@ -0,0 +1,95 @@
+---
+# "devbind" role binds network device to the requested kernel driver.
+# Usage: set "pci_address" and "driver" variables
+#  - name: use devbind
+#    include_role:
+#      name: devbind
+#    vars:
+#      driver: uio_pci_generic
+#      pci_address: "{{ item }}"
+#    with_items:
+#      - "0000:02:0f.6"
+#      - "0000:02:0f.7"
+
+- name: get PCI device details
+  block:
+  - name: get PCI device details
+    shell: lspci -vmmks {{ pci_address }}
+    register: lspci
+
+  - name: parse PCI device details
+    set_fact:
+      dev_info: "{{ lspci.stdout | replace('\t',' ') | from_yaml }}"
+      interfaces: []
+
+  - name: find network interfaces associated with the device
+    find:
+      paths: '{{ "/sys/bus/pci/devices/" + pci_address + "/net" }}'
+      recurse: no
+      file_type: any
+    register: ls
+
+  - name: create list of network interfaces names
+    set_fact:
+      interfaces: "{{ interfaces + [(item | basename)] }}"
+    with_items: "{{ ls.files | map(attribute='path') | list }}"
+
+  - name: add network interfaces names to device info
+    set_fact:
+      dev_info: "{{ dev_info | combine({ 'Interfaces': interfaces }) }}"
+
+- name: fail if network device is active in the routing table
+  block:
+  - name: get routing table
+    shell: ip -o route
+    register: ip_route
+
+  - name: filter link local routes
+    set_fact:
+      routes: "{{ ip_route.stdout_lines | reject('search', '169.254') | list }}"
+
+  - name: assert that none of the interfaces are active in the routing table
+    assert:
+      that: "routes | select('search', item) | list | length == 0"
+      fail_msg: "{{ item }} is active in the routing table, refusing further actions"
+      success_msg: "{{ item }} isn't active in the routing table, continuing..."
+    with_items: "{{ dev_info.Interfaces }}"
+
+  - name: display device information
+    debug: var=dev_info
+
+- name: bind device to the driver
+  block:
+  - name: ensure that kernel module is loaded
```

Comment:
Also, lets do this earlier on instead of inside the driver binding loop. We check if the driver module exist before we make call to the devbind role setting the "driver" var.